packages/flutter/lib/src/widgets/interactive_viewer.dart

```diff
@@ -923,7 +924,22 @@ class _InteractiveViewerState extends State<InteractiveViewer> with TickerProvid
         _transformationController.value,
         focalPointSceneScaled - focalPointScene,
       );
+      final double scale = _transformationController.value.getMaxScaleOnAxis();
```

Comment:
I think scaleChange might be what we want here.  In _onScaleUpdate, it calls onInteractionUpdate with the direct `details.scale` value, which doesn't consider the current scale of the child.

Reply:
I'm not sure about this one because it seems that in the normal pinch zooming it seems to display the currentScale instead of the scaleChange, could you do a crosscheck on this just to make sure. Also the ScaleChange value seems to be fixed at approximately 1.3 for zoom in and 0.89 for zoom out so I'm not sure which would be the better choice. I'm giving my opinion on the decision but do update me on which one you think should be used  for better change!

Reply:
I also have a follow up question regarding the flutter_build check, it seems there are times when it was fine and times it is not, is this a normal thing?

Reply:
Regarding the 1.3/0.89 thing, is your InteractiveViewer wrapped in a GestureDetector somewhere?  There is a known issue about that which I still need to fix: https://github.com/flutter/flutter/issues/58636

Regarding the current scale vs. scaleChange, it does seem to be passing the change in scale based on my experiment:

```dart
InteractiveViewer
  onInteractionUpdate: (ScaleChangeDetails details) {
    print('onInteractionUpdate ${details.scale}');
  },
  ...
)
```

Logs:
```
(Zooming in:)
flutter: onInteractionUpdate 1.0051666615057695
flutter: onInteractionUpdate 1.0103338151812797
...
flutter: onInteractionUpdate 2.4213762341047356
(Stopping and zooming out:)
flutter: onInteractionUpdate 0.9978297973338419
flutter: onInteractionUpdate 0.995660468727219
...
```

Lastly, about the flutter_build check.  Yes, our test infrastructure is quite flakey unfortunately.  If the error message doesn't seem related to your PR, then it's probably an infrastructure problem and just needs to be re-run.

Reply:
the 1.3/0.89 I think is becaused a scroll tick is always constant value, so probably why the scaleChange value is constant. But since your tests are proven scaleChange, then I will be using that for the update!